src/qml/qgismobileapp.qml

```diff
@@ -2236,8 +2243,37 @@ ApplicationWindow {
       }
 
       Connections {
+        id: mapSettingsConnections
         target: mapCanvas.mapSettings
 
+        property bool autoReFollowLocation: false
+
+        function cancelAutoFollowLocation() {
+          toast.close();
+          mapSettingsConnections.autoReFollowLocation = false;
+        }
+
+        function followLocation() {
+          if (mapSettingsConnections.autoReFollowLocation) {
+            gnssButton.followActive = true;
+            mapCanvasMap.freeze('follow');
+            if (positionSource.projectedPosition.x) {
+              if (!positionSource.active) {
+                positioningSettings.positioningActivated = true;
+              } else {
+                gnssButton.followLocation(true);
+              }
+            } else {
+              if (positionSource.valid) {
+                if (positionSource.active) {
+                } else {
+                  positioningSettings.positioningActivated = true;
+                }
+              }
+            }
+          }
+        }
```

Comment:
Let's not do that, instead just define the function within the displayToast() call itself.